moveit_core/robot_model/src/joint_model.cpp

```diff
@@ -54,8 +56,6 @@ JointModel::JointModel(const std::string& name)
   , mimic_offset_(0.0)
   , passive_(false)
   , distance_factor_(1.0)
-  , first_variable_index_(-1)
```

Comment:
btw, I very much prefer having explicit indexes enforced with the constructor over allowing constructing stand-alone joints in an invalid state.